src/lib.rs

```diff
@@ -422,7 +422,82 @@
 //!
 //! - `#[feature(untagged_unions)]` for overlapping/overloaded registers
 
-// NOTE This file is for documentation only
+#![recursion_limit = "128"]
+
+extern crate cast;
+extern crate either;
+#[macro_use]
+extern crate error_chain;
+extern crate inflections;
+#[macro_use]
+extern crate quote;
+extern crate svd_parser as svd;
+extern crate syn;
+
+mod errors;
+mod generate;
+mod util;
+
+pub use util::Target;
+
+pub struct Generation {
+    pub lib_rs: String,
+    pub device_specific: Option<DeviceSpecific>,
+
+    #[allow(dead_code)]
+    reserved: (),
```

Comment:
To match the style of svd-parser, I'd want this to be changed to

```
// Reserve the right to add more fields to this struct
_extensible: (),
```

That way we don't use an unnecessary allow and clarify (in simple terms) why we do this